### PR TITLE
Made code changes as per issue#1145 

### PR DESCRIPTION
**Pull request by yugander-krishan-singh**
- Make the following Artifact Generators implement the `ArtifactGenerator` interface
    - `NpmPackageGenerator.java`
    - `NugetPackageGenerator.java`
    - `MavenArtifactGenerator.java`

### CODE REVIEW

**Comment by carlspring** [comment]
Hi!

Thank you for your contribution!

Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

Also, please, feel free to join our [chat channel](https://chat.carlspring.org/), if you'd like to learn more about the project and/or like to find out what else you could help with.

Kind regards,

Martin

**Comment by carlspring** [comment]
@yugander-krishan-singh ,

The username under which you're committing + pushing to Github seems to be mis-configured. You will have to look into how to configure your git settings properly and amend the author of the commits and then force push back to the branch. (When your user is properly set up, you see your avatar when against your commits in the commit history, otherwise it's a default grey avatar).

**Comment by yugander-krishan-singh** [comment]
@carlspring I did change the author to my user-id as mentioned in the above message but it seems that committer is still the old username. Is it fine or should I try changing the committer also?

Also I have mailed the ICLA as mentioned in the [contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/MavenArtifactGenerator.java [hunk 1]
```diff
@@ -316,4 +318,11 @@ public void setBasedir(String basedir)
         this.basedir = basedir;
     }
 
+    @Override
+    public Path generateArtifact(URI uri,
+                                 int size)
+            throws IOException
+    {
+        return null;
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/MavenArtifactGenerator.java]: I think that these methods need to be implemented.

Comment (yugander-krishan-singh) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/MavenArtifactGenerator.java]: Thanks for letting me know. Actually issue [#1145](https://github.com/strongbox/strongbox/issues/1145) description says -  We need to make all this classes implement the` org.carlspring.strongbox.artifact.generator.ArtifactGenerator` interface.
Now after doing the above I did think about whether I need to do the implementation. For Example: overridden method `generateArtifact` takes 2 args `URI uri, int size` but I am not sure how to use them.

**Comment by yugander-krishan-singh** [comment]
@carlspring  closing the PR as code implementation is not complete.

**Comment by carlspring** [comment]
No need to close it. You can still push further improvements to it, if you'd like.

**Pull request closed without merging**